CDCgov/prime-reportstream
# Breadcrumbs on Submissions Detail don't link back to previous URL

This PR adds context to the link state for submission reports on the message tracker page and a new previousPage prop to Crumbs so that breadcrumbs can display a "Return to Message Tracker ID Search" button on top of the existing breadcrumbs for the submissions sitemap area. Minor updates were made to the Table component to support passing link state.

Test Steps:
1. Login
2. Under the Admin dropdown in navigation, click Feature Flags
3. Add message-tracker to your active Feature Flags
4. Under the Admin dropdown in navigation, click Message ID Tracker
5. Search (ex: 014228, 798189, etc.)
6. The message will appear in the results, click on the Report ID
7. Verify that the breadcrumbs shows a back button to Message Tracker on top of the breadcrumbs

## Changes
- "from" property added to link state for reports on message-tracker
- Table component and related types updated to support passing link state

## Checklist

### Testing

### Process

## Linked Issues
- Fixes #7712
